crates/sql_provider_datafusion/src/dbconnection.rs

```diff
@@ -0,0 +1,22 @@
+use ::duckdb::arrow::array::RecordBatch;
+use datafusion::{arrow::datatypes::SchemaRef, sql::TableReference};
+use snafu::prelude::*;
+
+pub mod duckdbconn;
+
+#[derive(Debug, Snafu)]
+pub enum Error {
+    #[snafu(display("DuckDBError: {source}"))]
+    DuckDBError { source: duckdb::Error },
```

Comment:
nit. if we put DuckDBError here, for any new type of implementation, we will need to put PostgresError here even we decided to implement this trait outside the crate.

Maybe wrapped it into a dyn Error?

Reply:
Good point

Reply:
Actually, this caused issues using the crate because it doesn't implement Send. I think it also makes the errors harder to work with

Reply:
got this working, done.